concrete/controllers/panel/detail/page/composer.php

```diff
@@ -60,6 +60,21 @@ public function saveAndExit()
         }
     }
 
+    public function justSave()
+    {
+        if ($this->validateAction()) {
+            $r = $this->save();
+            $ptr = $r[0];
+            $u = new User();
+            $u->unloadCollectionEdit();
+            $c = $this->page;
+            $ptr->setRedirectURL(Core::make('helper/navigation')->getLinkToCollection($c, true));
+            $ptr->outputJSON();
```

Comment:
Is there a way to reconfigure this to use response objects so that middleware can work here?

Reply:
To make it return a response rather than outputting directly so that middleware can interact with it:

eg:
```
return new JsonResponse($ptr);
```

instead of `$ptr->outputJSON()`